backend/app/db/session.py

```diff
@@ -0,0 +1,22 @@
+"""
+This module authenticates and connects to the neo4j database
+"""
+
+from py2neo import Graph
+from decouple import config
+
+# Grabbing environment variables
+HOSTNAME = config("DATABASE_HOST")
+USERNAME = config("DATABASE_USERNAME")
+PASSWORD = config("DATABASE_PASSWORD")
+
+
+def auth():
+    "Authenticating with the Database"
+    GRAPH = None
+    try:
+        GRAPH = Graph(HOSTNAME, auth=(USERNAME, PASSWORD))
+        print("Database authenticated")
+    except Exception as error:
+        print("Database authentication failed", error)
```

Comment:
This is a critical error - if we are unable to connect to database, the app should not start ideally. While I was running the initdb script, I saw that the script was trying to continue even when I didn't set the environment variables. it was throwing an error about NoneType, because it was trying to access functions inside GRAPH object, which is None. Ideally, we can print an error message that the environment variables are not set or even the above "Database authentication failed" error is fine, but the script should stop here.

Reply:
I'll raise an exception here